src/TypeSpec.Extension/Emitter.Csharp/src/lib/clientModelBuilder.ts

```diff
@@ -86,59 +86,51 @@ export function createModelForService(
         context,
         "@azure-tools/typespec-csharp"
     );
-    const title = service.title;
     const serviceNamespaceType = service.type;
-    const apiVersions: Set<string> = new Set<string>();
-    let version = service.version;
-    if (version && version !== mockApiVersion) {
-        apiVersions.add(version);
-    }
+
+    const apiVersions: Set<string> | undefined = new Set<string>();
+    let defaultApiVersion: string | undefined = undefined;
     const versions = getVersions(program, service.type)[1]?.getVersions();
     if (versions) {
         for (const ver of versions) {
             apiVersions.add(ver.value);
         }
-        version = versions[versions.length - 1].value; //default version
+        defaultApiVersion = versions[versions.length - 1].value;
     }
+    const apiVersionParam: InputParameter | undefined = defaultApiVersion
```

Comment:
Is it possible to add some unit test to verify the behavior of api-version in different cases?